ocean_lib/exceptions.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+
+class ContractNotFound(Exception):
+    """
+    Contract address is not found
+    in the factory events
+    """
+
+
+class DDOError(Exception):
+    """
+    Generic DDO Error
+    """
+
+
+class InsufficientBalance(Exception):
+    """
+    The token balance is not sufficient
+    """
+
+
+class AssetsError(Exception):
+    """
+    Generic Assets Error
+    """
+
+
+class BPoolError(Exception):
+    """
+    Generic Balancer Pool Error
+    """
```

Comment:
We can delete this, since all the places it's used below can be replaced by VerifyTxFailed